dm/syncer/syncer.go

```diff
@@ -2967,6 +2987,55 @@ func (s *Syncer) trackDDL(usedSchema string, trackInfo *ddlInfo, ec *eventContex
 	return nil
 }
 
+func (s *Syncer) trackOriginDDL(ev *replication.QueryEvent, ec eventContext, originSQL string) error {
```

Comment:
seems we can get `originSQL` by use `strings.TrimSpace(string(ev.Query))`

Reply:
btw most code here is same as `handleQueryEvent` can we try to merge them?

Reply:
I have thought about this before, but finally, I choose to rewrite a new method.
Because many places in `handleQueryEvent` may return an error before skipping. If we got an error before we skip this ddl, we can never skip this ddl through `binlog skip`.

Reply:
Now it seems that no matter whether this DDL is executed or not, as long as this DDL can be tracked(not build skip ddl, can be parsed, can be split, can genDDLInfo), we need to track it.

So, how about move all `TrackDDL` to a unified place, maybe before `binlog skip`.

Do you think this is feasible?

Reply:
Could you please show me which place is `before binlog skip`?

Reply:
This `TrackDDL` seems can't unify with others(at least I didn't get any methods), because it didn't report error as others.

The others `TrackDDL` probably can unify(to be discussed further), but is is another problem. We needn't  talk it in this pr.